scripts/update_project_mailing_lists.py

```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8 -*-
+""" Cron script that attempts to ensure synchronicity between
+    remote mailing lists on MailGun and their local representations.
+"""
+import logging
+
+from modularodm import Q
+
+from framework.celery_tasks import app as celery_app
+from framework.transactions.context import TokuTransaction
+
+from website.app import init_app
+from website.mailing_list.utils import full_update
+from website.project.model import Node
+
+logger = logging.getLogger(__name__)
+
+def main(dry_run=True):
+    init_app(routes=False)
+    updated_nodes = Node.find(Q('mailing_updated', 'eq', True))
+    for node in updated_nodes:
+        update_node(node, dry_run)
+
+    if dry_run:
+        raise RuntimeError('Dry run, transaction rolled back.')
+
+def update_node(node, dry_run=True):
+    if dry_run:
+        def full_update(*args, **kwargs):
+            # Override actual method to prevent outgoing calls
+            return
+
+    # reload the node to ensure that it is current
+    node.reload()
+
+    assert not node.is_registration and not node.is_collection
+
+    # Reset mailing_updated now in case of a user-change during this update
+    node.mailing_updated = False
```

Comment:
As discussed on flowdock, there may be a case where another celery task triggers an action that needs to be done, while this script is running.

This could trigger an edge case where mailgun and the db were not consistent; one solution would be for the short-lived celery tasks to update the DB record when they fail, to indicate work needs to be done to clean up after a problem.

Will discuss more in person. I'm flying a bit blind without a clear sense of all the interconnected failure modes these tasks are designed to handle.